lib/stats/stats-query.c

```diff
@@ -182,6 +192,95 @@ _query_counter_hash(gchar *key_str)
   return counters;
 }
 
+static GList *
+_get_input_counters_of_view(ViewRecord *view)
+{
+  GList *counters_of_view = NULL;
+  for (GList *q = view->queries; q; q = q->next)
+    {
+      GList *selected_counters = NULL;
+      gchar *query = q->data;
+
+      selected_counters = _query_counter_hash(query);
+      if (selected_counters == NULL)
+        continue;
+
+      counters_of_view = g_list_concat(counters_of_view, selected_counters);
+    }
+  return counters_of_view;
+}
+
+static GList *
+_get_aggregated_counters_from_views(GList *views)
+{
+  GList *aggregated_counters = NULL;
+
+  for (GList *v = views; v; v = v->next)
+    {
+      ViewRecord *view = v->data;
+      GList *counters = _get_input_counters_of_view(view);
+
+      if (counters == NULL)
+        continue;
+
+      view->aggregate(counters, &view->counter);
+      aggregated_counters = g_list_append(aggregated_counters, view->counter);
+    }
+  return aggregated_counters;
+}
+
+static GList *
+_get_views(gchar *filter)
+{
+  GPatternSpec *pattern = g_pattern_spec_new(filter);
+  GList *views = NULL;
+  gpointer key, value;
+  GHashTableIter iter;
+  gboolean single_match;
+
+  single_match = _is_single_match(filter);
+
+  g_hash_table_iter_init(&iter, stats_views);
+  while (g_hash_table_iter_next(&iter, &key, &value))
+    {
+      if (_is_pattern_matches_key(pattern, key))
+        {
+          ViewRecord *view = (ViewRecord *) value;
+          views = g_list_append(views, view);
+
+          if (single_match)
+            break;
+        }
+    }
+  g_pattern_spec_free(pattern);
+  return views;
+}
+
+static GList *
+_get_aggregated_counters(gchar *filter)
```

Comment:
The name `aggregated counter` is a good name: the counters that are assigned to an aggregated counters are existing independently.